src/main/java/org/spongepowered/api/util/command/source/RCONSource.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util.command.source;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Represents an Rcon client
+ */
+public interface RconSource extends RemoteSource {
+
+    /**
+     * Gets the {@link InetSocketAddress} used by the client to connect
+     * to the server.
+     *
+     * @return The {@link InetSocketAddress} used by the client
+     */
+    InetSocketAddress getAddress();
+
+    /**
+     * Gets whether this client is logged in, or authenticated.
+     *
+     * @return Whether this client is logged in
+     */
+    boolean getLoggedIn();
+
+    /**
+     * Sets whether this client is logged in, or authenticated.
+     *
+     * @param loggedIn Whether this client is logged in
+     */
+    void setLoggedIn(boolean loggedIn);
```

Comment:
This allows for plugins to do things like per-IP passwords, or allow multiple correct passwords.